docs/manual/docs/cmd/actions/completeworkflow.md

```diff
@@ -0,0 +1,83 @@
+# GitHub Actions workflow
+
+This file shows below and example of a workflow that can be used in a SPFx GitHub repo. Clicking on 'Actions' in a GitHub repo opens a YAML file. In that file the following code can be entered. 
+
+The code does the following
+* Builds the project
+* Bundles the solution
+* Creates the package
+* Uploads it to the artifacts
+* Downloads the package from artifacts
+* Uses Office CLI Login action to login to the tenant
+* Deploys the package using Office 365 CLI deploy action
+* Sends an email using the Office 365 CLI runscript action
+
+```
+name: SPFx CICD with O365 CLI
+
+on: [push]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        node-version: [10.x]
+    
+    steps:
+    # Checkout code
+    - name: Checkout code
+      uses: actions/checkout@v1
+      
+    # Setup node.js runtime
+    - name: Use Node.js ${{ matrix.node-version }}
+      uses: actions/setup-node@v1
+      with:
+        node-version: ${{ matrix.node-version }}
+    
+    # npm install
+    - name: Run npm install
+      run: npm install
+    
+    # gulp bundle and package solution
+    - name: Bundle and package
+      run: |
+        gulp bundle --ship
+        gulp package-solution --ship    
+    
+    # Upload artifacts (sppkg file)
+    - name: Upload artifact (sppkg file)
+      uses: actions/upload-artifact@v1.0.0
+      with:
+        name: output
+        path: sharepoint/solution/${{ secrets.sppkgFileName }}
+        
+  deploy:
+    needs: build
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        node-version: [10.x]
+    
+    steps:
+    
+    # Download package (sppkg file)
+    - name: Download pacakge (sppkg file)
+      uses: actions/download-artifact@v1
+      with:
+        name: output
+    
+    - name: Office 365 CLI Login
+      uses: pnp/office365-cli/actions/login@master
```

Comment:
Is this a reference to the Github repo .sh file directly?

If yes, what if the devs would like to add additional deployments steps and extend it with specifics to their build and release?
I am asking this question because there are few other options in my mind. What are we have a CLI command that will spawn these files in the SPFx solution '.github/workflows' folder so they can be referenced locally.
![image](https://user-images.githubusercontent.com/1049476/70230294-ec262700-174f-11ea-8be5-12589eb1b3f7.png)

Reply:
Yes, if we add `uses: pnp/office365-cli/actions/login@master` in the workflow then it will reference the GitHub action that will be in the master branch. That action will run in it's own container (no need to worry about that as developer) and do the required task.

Since this is a login action - the functionality of this is limited to login only. If more steps need to be executed then there are 2 options:

1. Use the 'runscript' action with which we can supply any number of Office 365 CLI commands
2. Directly run the Office 365 CLI commands in the workflow like I have done [here](https://gist.github.com/anoopt/756ca6dbfda79d3db797d6329c51600b)